File: Python/Euler004.py
#!/usr/bin/env python3
from __future__ import print_function
from itertools import combinations


def main():
    combos = combinations(range(100, 1000), 2)
    product_strings = (str(x * y) for x, y in combos)
    palindromes = (int(s) for s in product_strings if s == s[::-1])
    biggest = max(palindromes)
    print(biggest)


if __name__ == '__main__':
    main()

File: Ruby/Euler004.rb
#!/usr/bin/env ruby

def main
  combos = (100..999).to_a.combination(2)
  products = combos.map { |x, y| x * y }
  palindromes = products.filter { |x| x.to_s == x.to_s.reverse }
  biggest = palindromes.max
  puts biggest
end

if __FILE__ == $0
  main
end
